automated install/basic-install.sh

```diff
@@ -1754,44 +1770,78 @@ FTLinstall() {
     return 1
   fi
 
+  # Move into the temp ftl directory
+  pushd "$(mktemp -d)" > /dev/null || { echo "Unable to make temporary directory for FTL binary download"; return 1; }
+
   # Always replace pihole-FTL.service
   install -T -m 0755 "${PI_HOLE_LOCAL_REPO}/advanced/pihole-FTL.service" "/etc/init.d/pihole-FTL"
 
+  local ftlBranch
+  local url
+  local ftlBranch
+    
+  if [[ -f "/etc/pihole/ftlbranch" ]];then
+    ftlBranch=$(</etc/pihole/ftlbranch)
+  else
+    ftlBranch="master"
+  fi
+  
+  # Determine which version of FTL to download
+  if [[ "${ftlBranch}" == "master" ]];then
+    url="https://github.com/pi-hole/FTL/releases/download/${latesttag%$'\r'}"
+  else
+    url="https://ftl.pi-hole.net/${ftlBranch}"
+  fi
+
   # If the download worked,
-  if curl -sSL --fail "https://github.com/pi-hole/FTL/releases/download/${latesttag%$'\r'}/${binary}" -o "/tmp/${binary}"; then
+  if curl -sSL --fail "${url}/${binary}" -o "${binary}"; then
     # get sha1 of the binary we just downloaded for verification.
-    curl -sSL --fail "https://github.com/pi-hole/FTL/releases/download/${latesttag%$'\r'}/${binary}.sha1" -o "/tmp/${binary}.sha1"
-
-    # Move into the temp directory
-    cd /tmp
+    curl -sSL --fail "${url}/${binary}.sha1" -o "${binary}.sha1"
+    
+    # Make the tempory binary executable so that we can test the --resolver flag
+    chmod +x "${binary}"
+
+    # If the --resolver flag returns True (exit code 0), then we can safely stop & disable dnsmasq
+    if ./"${binary}" --resolver > /dev/null; then
+      if [[ $(which dnsmasq 2>/dev/null) ]]; then
+        stop_service dnsmasq
+        disable_service dnsmasq                
+        #ensure /etc/dnsmasq.conf contains `conf-dir=/etc/dnsmasq.d`
```

Comment:
Are we sure that this line is now always there? Assume there was never a `dnsmasq` install, then I think this would be skipped, no?

Reply:
The whole if block will be skipped if `dnsmasq` was not installed. However, `dnsmasq` _may` be pre-installed, and therefore we need to make sure that line exists